pkgs/development/libraries/harfbuzz/default.nix

```diff
@@ -10,6 +10,8 @@
 , meson
 , ninja
 , gobject-introspection
+, buildPackages
+, withIntrospection ? stdenv.hostPlatform.emulatorAvailable buildPackages
```

Comment:
oh this is a tricky pattern to get right! Is it documented somewhere to use the previous stage?

Reply:
I don't believe this function is documented at all, but this is how it's used in the few places it is.

Rather than documenting it though, I'd like to find a better interface.  I don't at all like how the implementation circumvents the lib/pkgs implementation by using a function that takes `pkgs` as its argument, so I think it should be moved into pkgs-lib or similar.  And there should be value that's always "can the host platform be emulated on the build platform?", because that's almost always what you want.  You shouldn't need to pass in package sets yourself for that.

Reply:
Fair enough, I didn't mean to imply something else should be done in their PR by my question anyways.